doc/stacks.md

```diff
@@ -0,0 +1,119 @@
+# Stacks
+
+Kubecf can be configured with support for multiple stacks. A stack consists of the operating system rootfs and a list of buildpacks.
+
+There are 2 stacks built in: The `cflinuxfs3` stack is defined by Cloud Foundry, and the `sle15` stack is based on SUSE Linux Enterprise.
+
+The `install_buildpacks` list in `values.yaml` determines which stacks will be installed when the kubecf helm chart is deployed. The first stack in the list will be the default stack:
+
+```
+install_buildpacks: [sle15, cflinuxfs3]
+```
+
+## Stack definitions
+
+A stack is defined in a single config file. A simplified example:
+
+```
+$ cat bclinux.yaml
+stacks:
+  bclinux:
+    description: "BigCorp Linux-based filesystem"
+    install_buildpacks: [bclang, java]
+
+    release_prefix: bigcorp
+    release_suffix: prod
+
+    releases:
+      '$defaults':
+        url: registry.bigcorp.com/kubecf
+      bclinux:
+        version: "1.5"
+      bigcorp-bclang-prod-buildpack:
+        version: "1.5.9.1"
+      bigcorp-java-prod-buildpack:
+        version: "4.32.1.1"
+```
+
+This describes the `bclinux` stack of BigCorp, supporting both their internal BigCorp language and Java.
+
+The `install_buildpacks` list describes the order in which the buildpacks are added to Cloud Foundry. The names in here are the short forms of the buildpack names, e.g. `bclang` stands for `bclang_buildpack`.
+
+The kubecf helm chart attempts to map release names to buildpack names automatically by stripping the `-buildpack` suffix first, and then stripping the optional `release_prefix` and `release-suffix`. If the remaining string then does not match the buildpack short name, then the release needs to include an explicit `buildpack` key:
+
+```
+    releases:
+      external_ruby_builder:
+        buildpack: ruby
+```
+
+## Modifying an existing stack
+
+In addition to defining additional stacks, it is also possible to modify the built-in stacks:
+
+```
+stacks:
+  cflinuxfs:
+    install_buildpacks: [nodejs]
+  sle15:
+    install_buildpacks_prepend: [ourjava]
+    install_buildpacks_append: [perl]
+    releases:
+      ourjava-buildpack:
+        version:1.0.0
+      suse-perl-buildpack:
+        version: 5.28.1
+```
+
+This will remove all buildpacks execept for `nodejs` from the `cflinuxfs3` stack, and add 2 buildpacks to the `sle15` stack, prepending `ourjava` to the front of the list, and appending `perl` to the end.
```

Comment:
> Would it be useful to be able to say "use the default buildpack list, but _drop_ python"?

I don't know, it would require yet another key to be added. I assumed the common case would be to add additional buildpacks to either the beginning or the end.

If you want to edit the list, you can just override `install_buildpacks` and list all the buildpacks in the order you want. I'm tempted to say that even the current prepend/append keys are borderline overkill.

Please either add a compelling use case, or "resolve" this conversation.

Reply:
I have no use case: I was curious because I couldn't think of a good reason for having _both_ append and prepend, and deletion seems like the missing one (with which you'd be able to reorder them, for some reason).

I'll just resolve this.